storage/transaction.go

```diff
@@ -20,6 +21,25 @@ func (s *Storage) GetTransactionWithBatchDetails(hash common.Hash) (
 	return transaction, nil
 }
 
+func (s *TransactionStorage) UpdateTransaction(tx models.GenericTransaction) error {
+	return s.executeInTransaction(TxOptions{}, func(txStorage *TransactionStorage) error {
+		txBase := tx.GetBase()
+		receipt, err := txStorage.getStoredTxReceipt(txBase.Hash)
+		if err != nil {
+			return err
+		}
+		if receipt == nil || receipt.ErrorMessage == nil {
+			return nil
+		}
```

Comment:
What do you think about returning meaningful errors instead returning nil? Right now UpdateTransaction can "fail silently" in some cases. I would return an error for each scenario and handle them in `updateDuplicatedTransaction`.

Reply:
Good idea 👍